client/src/components/HtModal/HtModal.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { ReactElement } from 'react';
+import {
+  Modal,
+  ModalBody,
+  ModalFooter,
+  ModalHeader,
+  ModalHeaderProps,
+  ModalProps,
+  ModalTitle,
+} from 'react-bootstrap';
+
+interface HtModalProps extends ModalProps {
+  showModal: boolean;
+  handleClose: () => void;
+  className?: string;
+}
+
+interface HtModalTitleProps extends ModalHeaderProps {
+  title: string;
+}
+
+interface HtModalHeaderProps extends ModalHeaderProps {
+  closeButton?: boolean;
+  children: ReactElement;
+}
+
+/**
+ * Modal with haztrak styling
+ * @param {{ showModal: boolean, handleClose: () => void, className?: string, children: ReactElement }} props react props
+ * @constructor
+ * @example
+ * <HtModal showModal={true} handleClose={() => {...}}>
+ *  <HtModal.Title title='sample title' />
+ * </HtModal>
+ */
+function HtModal({
+  showModal,
+  handleClose,
+  className,
+  children,
+  ...otherProps
+}: HtModalProps): ReactElement {
+  const baseAttributes = `bg-transparent ${className ? className : ''}`;
+  return (
+    <Modal
+      show={showModal}
+      onHide={handleClose}
+      className={baseAttributes}
+      {...otherProps}
+    >
+      {children}
+    </Modal>
+  );
+}
+
+/**
+ * Modal Title for haztrak which acts as wrapper around react-bootstrap Modal Title
+ * @param {{title: string}} props react props
+ * @constructor
+ * @example
+ * <HtModal.Title title='sample title' />
+ */
+HtModal.Title = function ({ title, ...otherProps }: HtModalTitleProps): ReactElement {
```

Comment:
You Drilled down deeper than I did on these components ha. I just left the title as a string prop to be passed to Header. It's good to have this option, but I wonder if we can abstract the title details in the `HtModal.Header` props in the future. We can make that a new (low priority) issue.

Reply:
Wow great! Thanks for the feedback. I will work on this while fixing the redunant props issue I mentioned above